glib/settings_schema_source.go

```diff
@@ -50,15 +53,33 @@ func SettingsSchemaSourceNewFromDirectory(dir string, parent *SettingsSchemaSour
 	cstr := (*C.gchar)(C.CString(dir))
 	defer C.free(unsafe.Pointer(cstr))
 
-	return wrapSettingsSchemaSource(C.g_settings_schema_source_new_from_directory(cstr, parent.native(), gbool(trusted), nil))
+	schemaSource := C.g_settings_schema_source_new_from_directory(cstr, parent.native(), gbool(trusted), nil)
+	if schemaSource == nil {
+		return nil
+	}
+
+	//The finalizer is on the glib.SettingsSchemaSource.
+	s := wrapSettingsSchemaSource(schemaSource)
+	runtime.SetFinalizer(s, (*SettingsSchemaSource).Unref)
+
+	return s
 }
 
 // Lookup() is a wrapper around g_settings_schema_source_lookup().
 func (v *SettingsSchemaSource) Lookup(schema string, recursive bool) *SettingsSchema {
 	cstr := (*C.gchar)(C.CString(schema))
 	defer C.free(unsafe.Pointer(cstr))
 
-	return wrapSettingsSchema(C.g_settings_schema_source_lookup(v.native(), cstr, gbool(recursive)))
+	cSchema := C.g_settings_schema_source_lookup(v.native(), cstr, gbool(recursive))
+	if cSchema == nil {
+		return nil
+	}
+
+	//The finalizer is on the glib.SettingsSchema.
+	s := wrapSettingsSchema(cSchema)
+	runtime.SetFinalizer(s, (*SettingsSchema).Unref)
```

Comment:
Who calls `Ref()` here? simply adding `Unref()` may be bad.

Reply:
has "transfer full" on return -> i.e. don't Ref(), but ensure Unref() via finalizer